packages/xstate-react/src/useMachine.ts

```diff
@@ -74,36 +81,19 @@ export function useMachine<TMachine extends AnyStateMachine>(
   const service = useIdleInterpreter(getMachine, options as any);
 
   const getSnapshot = useCallback(() => {
-    if (service.status === InterpreterStatus.NotStarted) {
-      return (options.state
-        ? State.create(options.state)
-        : service.machine.initialState) as State<any, any, any>;
+    if (service.status === InterpreterStatus.NotStarted && options.state) {
+      const cached = cachedRehydratedStates.get(options.state);
+      if (cached) {
+        return cached;
+      }
+      const created = State.create(options.state) as State<any, any, any>;
+      cachedRehydratedStates.set(options.state, created);
+      return created;
     }
 
-    return service.state;
+    return service.getSnapshot();
   }, [service]);
 
-  const isEqual = useCallback(
-    (prevState, nextState) => {
-      if (service.status === InterpreterStatus.NotStarted) {
-        return true;
-      }
-
-      // Only change the current state if:
-      // - the incoming state is the "live" initial state (since it might have new actors)
-      // - OR the incoming state actually changed.
-      //
-      // The "live" initial state will have .changed === undefined.
-      const initialStateChanged =
-        nextState.changed === undefined &&
-        (Object.keys(nextState.children).length > 0 ||
-          typeof prevState.changed === 'boolean');
```

Comment:
We no longer need this because actors are now lazy and we are not recomputing initial state when starting etc